platform/mellanox/docker-syncd-mlnx/ecmp_calculator/ecmp_calc_sdk.py

```diff
@@ -0,0 +1,448 @@
+#!/usr/bin/env python3
+
+from python_sdk_api.sx_api import *
+from port_utils import sx_check_rc, SWITCH_ID
+import struct
+import socket
+import ctypes
+import sys
+
+PORT='port'
+VPORT='vport'
+VLAN='vlan'
+SX_ENTRY_NOT_FOUND = -1
+SDK_DEFAULT_LOG_LEVEL=SX_VERBOSITY_LEVEL_ERROR
+
+def severity_to_verbosity(severity):
+    value = severity + 1
+    verbosity = 0
+    while not (value & 1):
+        value = value >> 1
+        verbosity = verbosity + 1
+    return verbosity
+
+def log_cb(severity, module_name, msg):
+    verbosity = severity_to_verbosity(severity)
+    log_level = {SX_VERBOSITY_LEVEL_INFO: 'LOG_INFO', SX_VERBOSITY_LEVEL_ERROR: 'LOG_ERR', 
+                 SX_VERBOSITY_LEVEL_WARNING: 'LOG_WARN', SX_VERBOSITY_LEVEL_DEBUG: 'LOG_DEBUG', 
+                 SX_VERBOSITY_LEVEL_NOTICE: 'LOG_NOTICE'}
+    if verbosity in log_level.keys():
+        level = log_level[verbosity]
+    else:
+        level = 'LOG'
+    if verbosity <= SDK_DEFAULT_LOG_LEVEL:
+        print('{}: {}'.format(level, ctypes.cast(msg,ctypes.c_char_p).value.decode('utf-8')))
+ 
+def sx_open_sdk_connection():
+    """ Open connection to SDK
+    
+    Args:
+        None
+    
+    Returns:
+        SDK handle
+    """     
+    sx_api = ctypes.CDLL('libsxapi.so', mode=ctypes.RTLD_GLOBAL)
+    log_cb_type = ctypes.CFUNCTYPE(None, ctypes.c_int, 
+                                   ctypes.POINTER(ctypes.c_char), 
+                                   ctypes.POINTER(ctypes.c_char))
+    log_cb_fn = log_cb_type(log_cb)
+    handle_p = ctypes.pointer(ctypes.c_int64())
+    
+    rc = sx_api.sx_api_open(log_cb_fn, handle_p)
```

Comment:
During cleanup before the application exits can we invoke sx_api_close?

Reply:
I added call to sx_api_close() in destructor.